### PR TITLE
fix: tests and typescript target

### PR DESCRIPTION
**Pull request by gunzip**
fix typescipt target in configuration file.

fixes test and linting errors.

it looks like i cannot install node-fetch with yarn (npm works):

```
4ef38" as pattern ["node-fetch@^2.1.2","node-fetch@^2.1.1"]. This could result in non-deterministic behavior, skipping.
error An unexpected error occurred: "https://github.com/bitinn/node-fetch: Invalid tar header. Maybe the tar is corrupted or it needs to be gunzipped?".
```

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Changes were made to package.json, but not to yarn.lock.<br/><i>Perhaps you need to run `yarn install`?</i>
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    

New dependencies added: <a href='https://www.npmjs.com/package/abort-controller'><code>abort-controller</code></a>.


<h2><a href="https://github.com/mysticatea/abort-controller#readme">abort-controller</a></h2>
<p>Author: Toru Nagashima</p>
<p>Description: An implementation of WHATWG AbortController interface.</p>
<p>Homepage: <a href="https://github.com/mysticatea/abort-controller#readme">https://github.com/mysticatea/abort-controller#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>8 months ago</td></tr><tr><td>Last Updated</td><td>2 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>4</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/event-target-shim'><code>event-target-shim</code></a></td></tr><tr><td>Keywords</td><td>w3c, whatwg, event, events, abort, cancel, abortcontroller, abortsignal, controller, signal and shim</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# abort-controller

[npm version](https://www.npmjs.com/package/abort-controller)
[Downloads/month](http://www.npmtrends.com/abort-controller)
[Build Status](https://travis-ci.org/mysticatea/abort-controller)
[Coverage Status](https://codecov.io/gh/mysticatea/abort-controller)
[Dependency Status](https://david-dm.org/mysticatea/abort-controller)

An implementation of [WHATWG AbortController interface](https://dom.spec.whatwg.org/#interface-abortcontroller).

```js
import AbortController from "abort-controller"

const controller = new AbortController()
const signal = controller.signal

signal.addEventListener("abort", () => {
    console.log("aborted!")
})

controller.abort()
```

> https://jsfiddle.net/1r2994qp/1/

## 💿 Installation

Use [npm](https://www.npmjs.com/) to install then use a bundler.

```
npm install abort-controller
```

Or download from [`dist` directory](./dist).

- [dist/abort-controller.mjs](dist/abort-controller.mjs) ... ES modules version.
- [dist/abort-controller.js](dist/abort-controller.js) ... Common JS version.
- [dist/abort-controller.umd.js](dist/abort-controller.umd.js) ... UMD (Universal Module Definition) version. This is transpiled by [Babel](https://babeljs.io/) for IE 11.

## 📖 Usage

```js
import AbortController from "abort-controller"
// or
const AbortController = require("abort-controller")

// or UMD version defines a global variable:
const AbortController = window.AbortControllerShim
```

### AbortController

> https://dom.spec.whatwg.org/#interface-abortcontroller

#### controller.signal

The [AbortSignal](https://dom.spec.whatwg.org/#interface-AbortSignal) object which is associated to this controller.

#### controller.abort()

Notify `abort` event to listeners that the `signal` has.

## 📰 Changelog

- See [GitHub releases](https://github.com/mysticatea/abort-controller/releases).

## 🍻 Contributing

Contributing is welcome ❤️

Please use GitHub issues/PRs.

### Development tools

- `npm install` installs dependencies for development.
- `npm test` runs tests and measures code coverage.
- `npm run clean` removes temporary files of tests.
- `npm run coverage` opens code coverage of the previous test with your default browser.
- `npm run lint` runs ESLint.
- `npm run build` generates `dist` codes.
- `npm run watch` runs tests on each file change.


</details>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: package.json [hunk 1]
```diff
@@ -24,24 +24,25 @@
...
-    "ts-jest": "^22.4.2",
+    "ts-jest": "^22.4.6",
     "tslint": "^5.1.0",
     "tslint-config-prettier": "^1.6.0",
     "tslint-immutable": "^4.5.4",
     "tslint-plugin-prettier": "^1.3.0",
     "tslint-sonarts": "^1.6.0",
     "typedoc": "^0.11.1",
-    "typescript": "^2.8.0",
+    "typescript": "~2.8.4",
```
Comment (cloudify) [inline package.json]: why?

Comment (gunzip) [reply package.json]: otherwise yarn will update to 2.9 which is not compatible with the last version of io-ts

Comment (cloudify) [reply package.json]: yarn doesn't update if you don't update yarn.lock :)

Comment (gunzip) [reply package.json]: I cant because I cannot install node-fetch package (see above) and yarn breaks there (that's why we cannot merge this PR : )

**Pull request merged**